### PR TITLE
Add new FEVM actors to the registry

### PR DESCRIPTION
**Pull request by arajasek**
## Related Issues

Should fix https://github.com/filecoin-project/ref-fvm/issues/1022

## Proposed Changes

- Add the missing actors to the registry
- Skip actors with nil state
- Stop special-casing the account actor -- I don't think we ever should have been doing this?

## Additional Info

Needs https://github.com/filecoin-project/go-state-types/pull/143 in a release

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Review by geoff-vball — APPROVED** [review]

File: chain/vm/invoker.go [hunk 1]
```diff
@@ -284,17 +284,16 @@ func DecodeParams(b []byte, out interface{}) error {
...
 	actInfo, ok := i.actors[act.Code]
 	if !ok {
 		return nil, xerrors.Errorf("state type for actor %s not found", act.Code)
 	}
 
 	um := actInfo.vmActor.State()
+	if um == nil {
+		// TODO: I would like to assert that we have the empty object here
+		return nil, nil
+	}
```
Comment (magik6k) [inline chain/vm/invoker.go:296]: Do the embryo actors actually not have state? What does the state cid in actor entry point to?

Comment (arajasek) [reply chain/vm/invoker.go:296]: Embryo actors (soon to be renamed to "placeholders"), EthAccounts, and the EAM do not have state. Their `head`s are the [empty object CID](https://github.com/filecoin-project/lotus/blob/0b6d1ad20bd629c9b309262339d400ece81d3840/chain/vm/mkactor.go#L40).

This CID will get put into the blockstore as part of [the migration](https://github.com/filecoin-project/go-state-types/blob/08baf7047877664962ecfefdae09a4f4c7f4741f/builtin/v10/migration/top.go#L354-L361) (if it isn't already).

File: chain/actors/builtin/registry.go [hunk 2]
```diff
@@ -343,6 +348,32 @@ func MakeRegistry(av actorstypes.Version) []RegistryEntry {
...
+				registry = append(registry, RegistryEntry{
+					code:    codeID,
+					methods: evm10.Methods,
+					state:   new(evm10.State),
+				})
+			case manifest.EamKey:
+				registry = append(registry, RegistryEntry{
+					code:    codeID,
+					methods: eam10.Methods,
+					state:   nil,
```
[the remark this answers is not in the document]
Comment (arajasek) [reply chain/actors/builtin/registry.go:362]: @geoff-vball Continue convo [here](https://github.com/filecoin-project/lotus/pull/9958#discussion_r1061735660)?

File: chain/vm/invoker.go (hunk 1 above)
Comment (magik6k) [reply chain/vm/invoker.go:296]: Then I guess, if we’re not using nil state anywhere to mean “actor not found”, this seems fine

Comment (arajasek) [reply chain/vm/invoker.go:296]: Good question -- I don't know of any place where we do that, but I'll take a look before assuming.

**Pull request merged**